### PR TITLE
Added autofill observer demo

### PR DESCRIPTION
**Pull request by maudnals**
This demo showcases how to programmatically determine whether a user has used browser autofill to fill form fields, or whether the user has manually filled the field.

<p align="center">
  
</p>

### CODE REVIEW

File: autofill-observer-demo/address.html [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+      </nav>
+      <h1>Address</h1>
+      <div class="data-warning">
+        This demo does not store or send any data to a server. 
+      </div>
+      <form id="form" onsubmit="return submitForm(event)">
+        <div>
+            <label for="name">Name</label>
+            <input type="text" id="name" name="name" autocomplete="name" enterkeyhint="next">
+          </div>
```
Comment (battre) [inline autofill-observer-demo/address.html:68]: super nit: indentation -2?

File: autofill-observer-demo/script.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  allFieldsAsArray.forEach((fieldEl) => {
+    fieldEl.addEventListener('change', function (event) {
+      updateAutofillStatus(formEl, fieldEl);
+    });
+  });
+}
+
+function getAllAutofilledFields(formEl) {
+  // Get all elements that are autofilled, using the :autofill pseudo-class
+  return formEl.querySelectorAll(':is(:autofill)');
```
Comment (battre) [inline autofill-observer-demo/script.js:46]: I am by no means a CSS guru. It looks to me like you need the `:is` only if you want to combine multiple selectors. So maybe it makes sense to replace `:is(:autofill)` in all documents by `:autofill`.

File: autofill-observer-demo/address.html [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        <h2>Autofill info</h2>
+        <div id="autofillInfo">
+          To see autofill info, fill the fields and click the button.
+        </div>
+      </div>
+    </main>
+    <script type="module" src="./script.js"></script>
+  </body>
+</html>
+
```
Comment (battre) [inline autofill-observer-demo/address.html:348]: nit: delete these blank linkes?

File: autofill-observer-demo/script.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+// Global variable storing autofill statuses for each field
+const autofillStatuses = {};
+// Example: {
+//     "name": "autofilled",
+//     "street-address": "never-autofilled",
+//     "country": "autofilled-then-fixed"
+// }
+
+function initializeAutofillObserver(formEl) {
```
Comment (battre) [inline autofill-observer-demo/script.js:30]: Should this be initializeChangeObservers?

**Review by battre — APPROVED** [review]
Cool! All optional comments.

TBH: I am not an HTML/CSS/JS guru.

File: autofill-observer-demo/address.html (hunk 1 above)
Comment (maudnals) [reply autofill-observer-demo/address.html:68]: Fixed, ty - Formatter issue

File: autofill-observer-demo/address.html (hunk 3 above)
Comment (maudnals) [reply autofill-observer-demo/address.html]: Fixed, ty - Formatter issue

File: autofill-observer-demo/script.js (hunk 4 above)
Comment (maudnals) [reply autofill-observer-demo/script.js]: Sounds good, renamed.

File: autofill-observer-demo/script.js (hunk 2 above)
Comment (maudnals) [reply autofill-observer-demo/script.js]: Good point! Replaced.

**Pull request merged**